tests/karma/unit/services/target-generator.js

```diff
@@ -338,4 +338,41 @@ describe('TargetGenerator service', function() {
     });
   });
 
+  it('updates targets that are no longer relevant - #3207', function(done) {
+    Settings.returns(KarmaUtils.mockPromise(null, { tasks: { targets: { items: [
+      { id: 'report', type: 'count' }
+    ] } } }));
+    UserContact.returns(KarmaUtils.mockPromise());
+    var callbackCount = 0;
+    injector.get('TargetGenerator')(function(err, actual) {
+      if (callbackCount === 0) {
+        chai.expect(actual.length).to.equal(1);
+        chai.expect(actual[0].id).to.equal('report');
+        chai.expect(actual[0].type).to.equal('count');
+        chai.expect(actual[0].count).to.equal(1);
+      } else if (callbackCount === 1) {
+        chai.expect(actual.length).to.equal(1);
```

Comment:
Targets are your monthly achievements such as "register 10 pregnancies". The count is how much progress you've made towards your target. These are shown on the Targets page even if you have made no progress towards your achievement yet. So in this case the target would show on the page with 0 pregnancies registered.